src/errors/ErrorWithStatus.ts

```diff
@@ -0,0 +1,12 @@
+export class ErrorWithStatus extends Error {
```

Comment:
Love this new addition -- minor organization point but it would be helpful to have it (and the related changes) in a separate commit (same PR is fine) since it represents an atomic improvement / not directly coupled with the deep query endpoint.

Reply:
I am rethinking this one because I probably gave feedback in the past that said "don't re-include an HTTP status in the body" and here I go including HTTP status in the body. The construct is nice but I hate seeing an HTTP status in a body, so even if I have to strip it out of the returned object I will. I agree it should be in a separate commit or PR. I am not entirely sold on squashing at all so I am still feeling my way through what should or should not be a separate commit (with regard to improvements) but your hint about atomicity makes good sense.

Reply:
I ditched the Error hierarchy changes as part of this PR because I would like to give it more thought when making it and also apply it more consistently/thoroughly in the codebase. In short, it might not happen haha.